pkg/core/async/queue.go

```diff
@@ -4,27 +4,35 @@ import (
 	"context"
 
 	"github.com/skygeario/skygear-server/pkg/core/config"
+	"github.com/skygeario/skygear-server/pkg/core/db"
 )
 
 type Queue interface {
-	Enqueue(name string, param interface{}, response chan error)
+	Enqueue(spec TaskSpec)
+	WillCommitTx() error
+	DidCommitTx()
```

Comment:
Embed the `TransactionHook` interface instead of duplicating its methods?

Reply:
I'm following Go's convention here of defining the interface only on consumer side, to avoid coupling provider interface with what consumer need.